WcaOnRails/app/models/user.rb

```diff
@@ -175,13 +173,6 @@ def name_must_match_person_name
     end
   end
 
-  validate :cannot_demote_senior_delegate_with_subordinate_delegates
-  def cannot_demote_senior_delegate_with_subordinate_delegates
-    if delegate_status_was == "senior_delegate" && delegate_status != "senior_delegate" && !subordinate_delegates.empty?
-      errors.add(:delegate_status, I18n.t('users.errors.senior_has_delegate'))
-    end
-  end
-
```

Comment:
Why are you dropping this validation? Even after this PR is merged, `delegate_status` field still exists, and `subordinate_delegates` also continues to exist (only the implementation has changed, but the accessor itself is still there). So why drop this?

Reply:
Thing is, since all delegates were connected to a senior delegate, we cannot demote a senior delegate without mapping all the subordinate delegates to the new senior delegate. The process that is done by board to change the senior delegate was as follows:
1. Change status of new senior delegate as 'senior delegate'.
2. Go to each subordinate delegates, and change the senior delegate to the new senior delegate
3. Now old senior delegate doesn't have any subordinate delegates, so demote them.

But now since it's connected to region, demoting a senior delegate won't affect the subordinate delegates since they are not tightly connected to the senior delegate. So now even if there are subordinate delegates, they can be demoted and the new senior delegate can be added after that. To get the freedom of removing senior delegate and adding new senior delegate, we are removing this validation.

If we need to add an alternative validation, it will be to make sure demotion of senior delegate doesn't take place unless there is another senior delegate for the region. But I think it will be better to implement that once the roles migration is completed.

Reply:
Okay, makes sense! Thanks for your thorough explanation.